src/hooks/use-tile-data-merging.ts

```diff
@@ -0,0 +1,31 @@
+import { useCallback } from "react";
+import {ITileLinkMetadata } from "../models/tiles/tile-link-types";
+import { ITileModel } from "../models/tiles/tile-model";
+import { useMergeTileDialog } from "./use-merge-data-dialog";
+import { mergeTwoDataSets } from "../models/data/data-set-utils";
+import { IDataSet } from "../models/data/data-set";
+import { getMergableDataSets, getTileDataSet } from "../models/shared/shared-data-utils";
+import { getSnapshot } from "@concord-consortium/mobx-state-tree";
+
+interface IProps {
+  model: ITileModel;
+}
+
+export const useTileDataMerging = ({model}: IProps) => {
+  const mergableTiles = getMergableDataSets(model);
```

Comment:
Should this be in a `useMemo`? As is, it seems like it will cause `useMergeTileDialog` to render a lot more times than necessary.